app/services/build_heritage.rb

```diff
@@ -16,7 +16,16 @@ def convert_params_for_model(new_params)
         service[:port_mappings_attributes] = service.delete(:port_mappings) if service[:port_mappings].present?
 
         unless service[:listeners].nil?
-          listener_map = Hash[Endpoint.where(name: service[:listeners].map { |e| e[:endpoint] }).pluck(:name, :id)]
+          endpoint_names = service[:listeners].map { |e| e[:endpoint] }
+
+          endpoints = if @district.present?
```

Comment:
@Yuuki77 Hmm. okay, I thought it would be null sometimes when this happens

Reply:
@davidsiaw 
in app/controllers/heritages_controller.rb:27
it is passed as nil
```ruby
  def update
    @heritage = BuildHeritage.new(permitted_params).execute
    @heritage.save_and_deploy!(without_before_deploy: false,
                               description: "Update to #{@heritage.image_path}")
    render json: @heritage
  end
```
 but I think since I updated the logic like this in app/services/build_heritage.rb:8
```ruby
    @district = district || @heritage.district
```
district cannot be nil so we can remove this condition.

Reply:
How about older data, before your change?

Reply:
@davidsiaw 
before it just assigned
`@district = district ` so I thought this can be nil in update.
but `@heritage.district` will not be nil in when heritage already exits